src/account/sync/input_selection.rs

```diff
@@ -17,29 +25,32 @@ pub fn select_input(target: u64, available_utxos: &mut Vec<u64>) -> crate::Resul
     }
 }
 
-fn single_random_draw(target: u64, available_utxos: &mut Vec<u64>) -> crate::Result<Vec<u64>> {
+fn single_random_draw(
+    target: u64,
+    available_utxos: &mut Vec<Address>,
```

Comment:
```suggestion
    available_utxos: &mut [Address],
```

Ditto, and probably all parameters with `&mut Vec<Address>` below.